src/Peachpie.Library/Spl/Datastructures.cs

```diff
@@ -271,71 +272,357 @@ public bool offsetExists(PhpValue offset)
     [PhpType(PhpTypeAttribute.InheritName), PhpExtension(SplExtension.Name)]
     public class SplDoublyLinkedList : Iterator, ArrayAccess, Countable
     {
-        public void __construct() => throw new NotImplementedException();
-        public virtual void add(PhpValue index, PhpValue newval) => throw new NotImplementedException();
-        public virtual PhpValue bottom() => throw new NotImplementedException();
-        public virtual int getIteratorMode() => throw new NotImplementedException();
-        public virtual bool isEmpty() => throw new NotImplementedException();
-        public virtual PhpValue pop() => throw new NotImplementedException();
-        public virtual void prev() => throw new NotImplementedException();
-        public virtual void push(PhpValue value) => throw new NotImplementedException();
-        public virtual string serialize() => throw new NotImplementedException();
-        public virtual void setIteratorMode(long mode) => throw new NotImplementedException();
-        public virtual PhpValue shift() => throw new NotImplementedException();
-        public virtual PhpValue top() => throw new NotImplementedException();
-        public virtual void unserialize(string serialized) => throw new NotImplementedException();
-        public virtual void unshift(PhpValue value) => throw new NotImplementedException();
+        protected readonly Context _ctx;
 
-        public virtual long count() => throw new NotImplementedException();
+        /// <summary>
+        /// SPL collections Iterator Mode constants
+        /// </summary>
+        [PhpHidden]
+        [Flags]
+        public enum SPL_ITERATOR_MODE
+        {
+            IT_MODE_KEEP = 0,
+
+            IT_MODE_FIFO = 0,
+
+            IT_MODE_DELETE = 1,
+
+            IT_MODE_LIFO = 2
+        }
+
+        public const int IT_MODE_KEEP = 0;
```

Comment:
good, just please keep the order of declarations as it is in PHP (so reflections will get the same result) and use the enum values instead of numbers:
```
IT_MODE_LIFO = (int)SPL_ITERATOR_MODE.IT_MODE_KEEP;;
IT_MODE_FIFO = ...
IT_MODE_DELETE = ...
IT_MODE_KEEP = ..
```